skbio/stats/distance/_base.py

```diff
@@ -1660,9 +1689,9 @@ def _build_results(
     )
 
 
-def _vec_to_size(vec_length: int) -> float:
+def _vec_to_size(vec: int) -> float:
     """Calculate the redundant size of a matrix given its condensed vector."""
-    return int((1 + np.sqrt(1 + 8 * vec_length)) / 2)
+    return int((1 + np.sqrt(1 + 8 * len(vec))) / 2)
```

Comment:
The return value is int whereas the function signature is float. Also, I wonder if the loss of floating-point precision can cause `np.sqrt(a**2) != a`. I tested a few and spotted none. But it will be great if we can get a firm answer from elsewhere that the current code is indeed the recommended way of calculating the size of a vector form matrix.